src/koreo_tooling/function_test.py

```diff
@@ -371,7 +391,7 @@ def _dict_compare(base_field: str, actual: dict, expected: dict) -> list[Compare
 
         key = f"{base_prefix}{mutual_key}"
 
-        if type(actual_value) != type(expected_value):
+        if not isinstance(actual_value, type(expected_value)):
```

Comment:
Are there any cases this could accidentally allow? It is _probably_ more accurate and correct, but I'm wondering if the use of type was to be more specific or just an accidental iteration.